app/src/gecko/java/org/mozilla/focus/web/WebViewProvider.java

```diff
@@ -283,6 +283,13 @@ public void onExternalResponse(GeckoSession session, GeckoSession.WebResponseInf
                     }
                 }
 
+                @Override
+                public void onCrash(GeckoSession session) {
+                    Log.i(TAG, "Crashed, reopening session");
+                    session.open(geckoRuntime);
+                    session.loadUri(currentUrl);
+                }
```

Comment:
Oh, is that save to do?

It would be awesome to send a custom event to Sentry (and maybe telemetry) here!

Reply:
New method, found the implementation in their example - https://dxr.mozilla.org/mozilla-central/source/mobile/android/geckoview_example/src/main/java/org/mozilla/geckoview_example/GeckoViewActivity.java#405

Reply:
Nice! Let's merge this and just file an issue for telemetry/sentry? :)